### PR TITLE
[animegrep.sh] use getopts

### LINKED ISSUE #5: Convert argument parsing to getopts
Currently how this is being done is too positional independent in my opinion. I think converting this to a `getopts` should allow us to set this properly.

### PR DESCRIPTION
**Pull request by sr229**
fixes GH-5

### CODE REVIEW

File: animegrep.sh [hunk 1]
```diff
@@ -17,43 +17,43 @@
...
+	echo ""
+	echo "-d <directory> sets a directory as your source media"
+	echo "-f <file> sets a file as your source media. You cannot set directory and files at the same time."
+	echo "-t <track> Not Documented. "
+	echo "-w <word> track number for your source media."
+	echo "-m to merge multiple output files as one. Useful if you have -d set."
+}
 
-# TODO: convert this to getopts so we don't have to set it the hackiest way possible.
 POSITIONAL=();
```
Comment (Emzi0767) [inline animegrep.sh:32]: Not POSIX-compatible.

File: animegrep.sh [hunk 2]
```diff
@@ -17,43 +17,43 @@
...
+	t) TRACK="${OPTARG}" ;;
+	w) WORD="${OPTARG}" ;;
+	m) MERGE=1 ;;
+	h) print_help ;;
+	*) print_help; exit 3 ;;
+   esac 
+done 
+shift $(($OPTIND -1))
+
 set -- "{$POSITIONAL[@]}";
```
Comment (Emzi0767) [inline animegrep.sh:52]: See L32.

Comment (sr229) [reply animegrep.sh:52]: POSIX compatibility is out of scope for this PR. I'll make a different branch for that.

File: animegrep.sh (hunk 1 above)
Comment (sr229) [reply animegrep.sh:32]: POSIX compatibility is out of scope for this PR. I'll make a different branch for that.

File: animegrep.sh [hunk 3]
```diff
@@ -17,43 +17,43 @@
...
+fi
+
+while getopts d:f:t:w:m:h opt; do
+  case "${opt}" in
+	d) DIRECTORY="${OPTARG}" ;;
+	f) FILE="${OPTARG}" ;;
+	t) TRACK="${OPTARG}" ;;
+	w) WORD="${OPTARG}" ;;
+	m) MERGE=1 ;;
+	h) print_help ;;
```
Comment (Emzi0767) [inline animegrep.sh]: If doing `-h`, might wanna do what most utils do and exit after printing usage.

**Pull request merged**